src/org/thoughtcrime/securesms/ConversationListAdapter.java

```diff
@@ -195,7 +195,12 @@ interface ItemClickListener {
   }
 
   void changeData(@Nullable DcChatlist chatlist) {
-    dcChatlist = chatlist == null ? new DcChatlist(0, 0) : chatlist;
+    if (chatlist == null) {
+      dcChatlist =  new DcChatlist(0, 0);
+    } else {
+      dcChatlist = chatlist;
+      dcContext = DcHelper.getContext(context.get());
```

Comment:
k, that is guaranteed be the same account as in the chatlist as the account is switched? 

still, we need to check context.get() for != null (this is the idea of a weak reference - the contained object may be freed, for whatever reason, usually this ConversationListAdapter object will be freed as well soonish):


```suggestion
    Context context = this.context.get();
    if (context == null) {
      return;
    }
    if (chatlist == null) {
      dcChatlist =  new DcChatlist(0, 0);
    } else {
      dcChatlist = chatlist;
      dcContext = DcHelper.getContext(context);
```

Reply:
> k, that is probably guaranteed be the same account as in the chatlist as the account is switched?

yes, at the time `changeData()` is called in `ConversationListFragment` the new DcChatlist that is passed is from the current selected account